### PR TITLE
calling .cuda on tokens before mask construction

### PR DESCRIPTION
**Pull request by SahilJain314**
# What does this PR do ?

free speedup on critic by doing mask construction on GPU

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def fwd_output_and_loss_func(data_iterator, model):
             mask = batch["mask"]
 
             attention_mask, _, position_ids = get_ltor_masks_and_position_ids(
-                tokens, self.tokenizer.eos_id, False, True, False,
+                tokens.cuda(), self.tokenizer.eos_id, False, True, False,
             )
 
             attention_mask = attention_mask[0:1].cuda(non_blocking=True)
```
Comment (odelalleau) [inline nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py:128]: I guess you can remove the `.cuda()` on l.128 now?

File: nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def fwd_output_and_loss_func(data_iterator, model):
             mask = batch["mask"]
 
             attention_mask, _, position_ids = get_ltor_masks_and_position_ids(
-                tokens, self.tokenizer.eos_id, False, True, False,
+                tokens.cuda(), self.tokenizer.eos_id, False, True, False,
```
Comment (gshennvm) [inline nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py:125]: I think it's better to just let the tokens be on cuda(i.e

```
tokens = batch["tokens"].cuda()
```

**Review by gshennvm — CHANGES_REQUESTED** [review]

File: nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py (hunk 2 above)
Comment (SahilJain314) [reply nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py]: tested, done

**Review by gshennvm — APPROVED** [review]
